components/Vote/ElectionCandidacy.js

```diff
@@ -0,0 +1,540 @@
+                          {vt('info/candidacy/delete')}
+                        </A>
+                      </div>
+                    }
+                  </Fragment>
+                )
+              }
+            </div>
+          </NarrowContainer>
+        } />
+      </Frame>
+    )
+  }
+}
+
+const cancelCandidacy = gql`mutation submitCandidacy($slug: String!) {
+  cancelCandidacy(slug: $slug) {
+    candidates {
+      id
+    }
+  }
+}`
+
+const updateCandidacy = gql`mutation updateCandidacy($slug:String!, $birthday: Date, $statement: String, $disclosures: String, $address: AddressInput, $portrait: String) {
+  submitCandidacy(slug: $slug) {
+    id
+    user {
+      id
+    }
+  }
+  updateMe(birthday: $birthday, statement: $statement, disclosures: $disclosures, address: $address, portrait: $portrait, hasPublicProfile: true) {
+    id
+  }
+}`
+
+const publishCredential = gql`
+  mutation publishCredential($description: String) {
+    publishCredential(description: $description) {
+      isListed
+      description
+    }
+  }
+
+`
+
+const query = gql`
+  query init {
+    election(slug: "${ELECTION_SLUG}") {
+      candidates {
+        id
+        yearOfBirth
+        city
+        recommendation
+        user {
+          id
+          name
+          username
+          email
+          statement
+          portrait
+          credentials {
+            isListed
+            description
+          }
+        }
+      }
+    }
+    me {
+      id
+      username
+      name
+      updatedAt
+      email
+      emailAccessRole
+      phoneNumber
+      phoneNumberNote
+      phoneNumberAccessRole
+      portrait
+      hasPublicProfile
+      isEligibleForProfile
+      statement
+      biography
+      disclosures
+      birthday
+      address {
+        line1
+        line2
+        postalCode
+        city
+        country
+      }
+      credentials {
+        isListed
+        description
+      }
+      publicUrl
+    }
+  }
+`
+
+export default compose(
+  withT,
+  voteT,
+  graphql(query),
+  graphql(publishCredential, {
+    props: ({mutate}) => ({
+      publishCredential: description => {
+        return mutate({
+          variables: {
+            description
+          }
+        })
+      }
+    })
+  }),
+  graphql(updateCandidacy, {
+    props: ({mutate, ownProps: {publishCredential, data: {me}}}) => ({
+      updateCandidacy: async (variables) => {
+        const credential = (me.credentials || []).find(c => c.isListed) || {}
+        if (variables.credential !== credential.description) {
+          await publishCredential(variables.credential || null)
+        }
+        return mutate({
+          variables,
+          refetchQueries: ['init']
+        })
+      }
+    })
+  }),
+  graphql(cancelCandidacy, {
+    props: ({mutate}) => ({
+      cancelCandidacy: slug => {
+        return mutate({
+          variables: {
+            slug
+          },
+          refetchQueries: ['init']
```

Comment:
use reference instead of name matching `refetchQueries: [{query}]`

Reply:
The reason I used the name was that refetchQueries don't seem to change the `loading` prop.

https://github.com/apollographql/apollo-client/issues/1263#issuecomment-279045988